web/src/components/LoginModal.tsx

```diff
@@ -0,0 +1,154 @@
+import { useEffect, useRef, useState } from 'react'
+import { setCookie } from '../lib/cookies'
+
+type LoginResponse = {
+  access_token: string
+  refresh_token: string
+}
+
+type Props = {
+  open: boolean
+  onClose: () => void
+  onSuccess: (payload: LoginResponse) => void
+}
+
+const API_BASE = import.meta.env.VITE_API_URL ?? ''
+
+export default function LoginModal({ open, onClose, onSuccess }: Props) {
+  const [email, setEmail] = useState('')
+  const [password, setPassword] = useState('')
+  const [error, setError] = useState<string | null>(null)
+  const [loading, setLoading] = useState(false)
+  const firstFieldRef = useRef<HTMLInputElement>(null)
+  const panelRef = useRef<HTMLDivElement>(null)
+
+  // Lock scroll when open
+  useEffect(() => {
+    if (!open) return
+    const prev = document.documentElement.style.overflow
+    document.documentElement.style.overflow = 'hidden'
+    return () => { document.documentElement.style.overflow = prev }
+  }, [open])
+
+  // Autofocus first field
+  useEffect(() => {
+    if (open) firstFieldRef.current?.focus()
+  }, [open])
+
+  // Close on ESC
+  useEffect(() => {
+    if (!open) return
+    const onKey = (e: KeyboardEvent) => { if (e.key === 'Escape' && !loading) onClose() }
+    window.addEventListener('keydown', onKey)
+    return () => window.removeEventListener('keydown', onKey)
+  }, [open, loading, onClose])
+
+  if (!open) return null
+
+  const submit = async (e: React.FormEvent) => {
+    e.preventDefault()
+    setError(null)
+    setLoading(true)
+    try {
+      const res = await fetch(`${API_BASE}/api/v1/auth/login`, {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify({ email, password }),
+      })
+      const json = (await res.json()) as LoginResponse | { message?: string }
+      if (!res.ok) {
+        const msg = (json as any)?.message || `Login failed (${res.status})`
+        throw new Error(msg)
+      }
+      const payload = json as LoginResponse      
+      setCookie('access_token', payload.access_token, { days: 1, sameSite: 'Lax' })
+      setCookie('refresh_token', payload.refresh_token, { days: 7, sameSite: 'Lax' })
```

Comment:
Cookies for access and refresh tokens are set without the secure flag which can expose tokens over HTTP in production. Add secure: true (conditional in dev) and consider httpOnly if set server-side.